File: API/Features/Billing/Receipts/Dtos/ReceiptReadDto.cs
using System;
using API.Features.Billing.Invoices;
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Billing.Receipts {

    public class ReceiptReadDto : IMetadata {

        // PK
        public string TransactionId { get; set; }
        // FKs
        public int CustomerId { get; set; }
        public int DocumentTypeId { get; set; }
        public int PaymentMethodId { get; set; }
        // Fields
        public DateTime Date { get; set; }
        public int InvoiceNo { get; set; }
        public decimal GrossAmount { get; set; }
        public string Remarks { get; set; }
        // Navigation
        public SimpleEntity Customer { get; set; }
        public DocumentTypeVM DocumentType { get; set; }
        public SimpleEntity PaymentMethod { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}

File: FrontEnd/src/app/features/billing/receipts/classes/dtos/form/receipt-read-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { DocumentTypeVM } from '../../view-models/list/documentType-vm'
import { Metadata } from 'src/app/shared/classes/metadata'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface ReceiptReadDto extends Metadata {

    // PK
    invoiceId: Guid
    // Fields
    date: string
    tripDate: string
    invoiceNo: number
    customer: SimpleEntity
    documentType: DocumentTypeVM
    paymentMethod: SimpleEntity
    grossAmount: number
    // Metadata
    postAt: string
    postUser: string
    putAt: string
    putUser: string

}
